### PR TITLE
feat(#928): D39-2.3a — frontend UI core skeleton (ADR-039 §6 Phase 2)

### PR DESCRIPTION
**Pull request by zjzcpj**
Closes #928

## Summary

ADR-039 §6 Phase 2 frontend skeleton — scaffolds the Git UI core (commit / history / diff / branches / status / stash) so D39-2.3b (impl) can drop in component bodies without re-reading ADR-039.

Every component carries a **≥30-line top-of-file docstring** covering state shape, props, layout markup with the exact `data-testid` selectors vitest tests will assert on, literal copy strings, keyboard shortcuts, accessibility, and edge cases. Component bodies `throw new Error("TODO: D39-2.3b — ...")` so vitest can detect skeleton state.

## What's in this PR

**NEW skeleton components** (`frontend/src/components/Git/`):
- `CommitDialog.tsx` — message editor + auto-detected file list + commit/cancel
- `GitHistoryList.tsx` — reverse-chrono list + Manual/All/Auto/Agent filter dropdown per ADR §3.4 / §3.4a / §3.5c
- `GitDiffModal.tsx` — react-diff-viewer-continued wrapper
- `BranchPicker.tsx` — toolbar dropdown with list/switch/create + Merge / Cherry-pick submenus
- `GitStatusBadge.tsx` — toolbar dirty/clean/conflict indicator
- `StashApplyDialog.tsx` — stash-on-restore prompt (literal copy from ADR §3.6)
- `StashListPanel.tsx` — Stash drawer (list/save/apply/drop)

**NEW slice** (`frontend/src/store/gitSlice.ts`):
- Default state for `branches / currentBranch / logCache / logLoading / historyFilter / status / stashes / mergeInProgress / lastError`.
- Synchronous setters (`setHistoryFilter / setMergeInProgress / setLastError / invalidateHistory`) are real.
- Async mutation actions (`loadBranches / loadLog / loadStatus / loadStashes / commit / switchBranch / createBranch / deleteBranch / restore`) throw the `TODO: D39-2.3b` error.
- **Pure helpers `classifyPrefix(message)` + `selectVisibleCommits(commits, filter)` are fully implemented + tested** — they encode ADR §3.4 / §3.5c filter semantics and are safe to consume in render.

**MODIFIED existing files** (mount slots / types / registration only):
- `frontend/src/types/api.ts` — TS interfaces matching PR #927 wire shapes (`GitCommit`, `GitBranch`, `GitStashEntry`, `GitDiff`, `GitStatus`, `GitMergeResult`, `GitCommitResponse`, `GitStashApplyResult`, `GitRestoreResult`, `GitHistoryFilter`, `GitCommitPrefix`).
- `frontend/src/lib/api.ts` — 18 typed function stubs wrapping `src/scieasy/api/routes/git.py`.
- `frontend/src/store/types.ts` — union `GitSlice` into `AppStore`.
- `frontend/src/store/index.ts` — register `createGitSlice` in the zustand factory.
- `frontend/src/hooks/useWebSocket.ts` — flip the `git.head_changed` TODO from `console.debug` to a real `useAppStore.getState().invalidateHistory()` call (wrapped in try/catch).
- `frontend/src/components/Toolbar.tsx` — mount a hidden `data-testid="git-toolbar-slot"` placeholder with an inline comment describing the exact layout D39-2.3b should drop in. **No behavioral wiring** — mounting the BranchPicker / GitStatusBadge components (whose bodies throw) would crash the toolbar on every render.

**NEW tests**:
- `Git/__tests__/CommitDialog.test.tsx` (13 cases — 6 pass for pure helpers, 7 `.skip` with test-plan docstrings)
- `Git/__tests__/GitHistoryList.test.tsx` (11 cases — 1 export check, 10 `.skip`)
- `Git/__tests__/BranchPicker.test.tsx` (11 cases — 1 export check, 10 `.skip`)
- `store/__tests__/gitSlice.test.ts` (20 cases — 18 pass for pure helpers + defaults + TODO-throw assertion, 2 `.skip`)

## Verification

- `cd frontend && npm run build` — clean tsc + vite build
- `vitest run` — **27 files, 224 passing, 42 skipped, 0 failing**
- No backend Python touched; no ADR-038 / Lineage file touched; no D39-2.4a-owned file (`MergeFlow`, `ConflictResolveView`, `ConflictMarkerDecoration`, `GitGraph/*`, `CodeEditor.tsx`) touched.

## Out of scope (explicitly NOT in this PR)

- Component implementations (D39-2.3b).
- Conflict resolution UI + GitGraph (D39-2.4a).
- Live Chrome smoke test — skeleton mounts only a hidden placeholder; first live smoke happens in D39-2.3b once components render real markup.

## Related Issues

Closes #928

🤖 Generated with Claude Code

### CODE REVIEW

File: frontend/src/types/api.ts [hunk 1]
```diff
@@ -236,3 +236,120 @@ export interface TreeEntry {
...
+  body: string;
+  /** Branch names whose tip is this commit (zero, one, or many). */
+  branches: string[];
+}
+
+/** Local branch row from `GET /api/git/branches`. */
+export interface GitBranch {
+  name: string;
+  /** Tip commit sha. */
+  commit_sha: string;
```
[the remark this answers is not in the document]
Comment (jiazhenz026) [reply frontend/src/types/api.ts]: accepted (fixed in 0b5e6b8) — renamed GitBranch.commit_sha to head_sha to match GitEngine.branches() in src/scieasy/core/versioning/git_engine.py.

File: frontend/src/types/api.ts [hunk 2]
```diff
@@ -236,3 +236,120 @@ export interface TreeEntry {
...
+}
+
+/** Stash entry from `GET /api/git/stash`. */
+export interface GitStashEntry {
+  /** Stash identifier (e.g. `stash@{0}`). */
+  stash_id: string;
+  /** Index in stash list (0 = most recent). */
+  index: number;
+  message: string;
+  created_at: string;
```
[the remark this answers is not in the document]
Comment (jiazhenz026) [reply frontend/src/types/api.ts]: accepted (fixed in 0b5e6b8) — replaced index+created_at with the actual wire fields (stash_id, message, date) to match GitEngine.stash_list() output.

File: frontend/src/types/api.ts [hunk 3]
```diff
@@ -236,3 +236,120 @@ export interface TreeEntry {
...
+}
+
+/**
+ * Result of `POST /api/git/merge` and `/cherry-pick`. The discriminated
+ * union mirrors `GitEngine.merge()` return classification — the backend
+ * normalizes git's exit code + stderr into one of these three shapes.
+ */
+export type GitMergeResult =
+  | { result: "fast-forward"; commit_sha: string }
+  | { result: "clean"; commit_sha: string }
```
[the remark this answers is not in the document]
Comment (jiazhenz026) [reply frontend/src/types/api.ts]: accepted (fixed in 0b5e6b8) — GitMergeResult success variants now carry conflicted_files: [] (no commit_sha), matching GitEngine.merge() / cherry_pick() which always return {result, conflicted_files}. Consumers needing the post-merge HEAD must call /api/git/log?limit=1 (documented in the JSDoc).
